libdnf/transaction/transaction_item.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright (C) 2017-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "libdnf/transaction/transaction_item.hpp"
+
+#include "libdnf/transaction/db/repo.hpp"
+#include "libdnf/transaction/transaction.hpp"
+#include "libdnf/transaction/transaction_item_action.hpp"
+#include "libdnf/utils/bgettext/bgettext-lib.h"
+
+
+namespace libdnf::transaction {
+
+
+std::string TransactionItem::get_action_name() {
+    return TransactionItemAction_get_name(action);
+}
+
+
+std::string TransactionItem::get_action_short() {
+    return TransactionItemAction_get_short(action);
+}
+
+
+TransactionItem::TransactionItem(Transaction & trans, Type item_type) : item_type{item_type}, trans{trans} {}
+
+
+bool TransactionItem::is_forward_action() const {
+    return TransactionItemAction_is_forward_action(action);
+}
+
+
+bool TransactionItem::is_backward_action() const {
+    return TransactionItemAction_is_backward_action(action);
+}
+
+
+/*
```

Comment:
Please could you add TODO with some description why you kept the code?

Reply:
I've added the TODO in the header file already.